lib/server.js

```diff
@@ -29,6 +29,9 @@ function handleResponse (opts, req, resp, response) {
         response.headers['Access-Control-Allow-Methods'] = 'GET';
         response.headers['Access-Control-Allow-Headers'] = 'accept, content-type';
 
+        // propagate the request id header
+        response.headers['X-Request-Id'] = req.headers['x-request-id'];
```

Comment:
You are right, there is no chance that any internal users is trying to read that header at this point.